sklearn/linear_model/_quantile.py

```diff
@@ -183,8 +192,19 @@ def fit(self, X, y, sample_weight=None):
                 f"Solver {self.solver} is only available "
                 f"with scipy>=1.6.0, got {sp_version}"
             )
+        else:
+            solver = self.solver
+
+        if solver not in (
+            "highs-ds",
+            "highs-ipm",
+            "highs",
+            "interior-point",
```

Comment:
Should we make this conditional on `scipy >= 1.11`?

Reply:
Agreed, we could imagine have an informative error message if you use `solver='interior-point'` but scipy>=1.11 (so `'interior-point'` has been removed from scipy)